src/dotnet-install.sh

```diff
@@ -816,39 +847,76 @@ install_dotnet() {
+        404)
+            say "The primary path $download_link is not available."
+            ;;
+        2*)
+            say "Cannot download via the primary path $download_link."
+            ;;
+        *)
+            say "Cannot download via the primary path $download_link. Returned HTTP status code: $primary_path_http_code."
+            ;;
+        esac
         rm -f "$zip_path" 2>&1 && say_verbose "Temporary zip file $zip_path was removed"
         if [ "$valid_legacy_download_link" = true ]; then
             download_failed=false
             download_link="$legacy_download_link"
             zip_path="$(mktemp "$temporary_file_template")"
             say_verbose "Legacy zip path: $zip_path"
             say "Downloading legacy link: $download_link"
-            download "$download_link" "$zip_path" 2>&1 || download_failed=true
 
+            # the download function will fill variables $http_code and $download_error_msg in case of failure.
+            http_code=""; download_error_msg=""
+            download "$download_link" "$zip_path" 2>&1 || download_failed=true
+            legacy_path_http_code="$http_code";  legacy_path_download_error_msg="$download_error_msg"
             if [ "$download_failed" = true ]; then
-                say "Cannot download: $download_link"
+                case $legacy_path_http_code in
+                404)
+                    say "The legacy path $download_link is not available."
+                    ;;
+                2*)
+                    say "Cannot download via the legacy path $download_link."
+                    ;;
+                *)
+                    say "Cannot download via the legacy path $download_link. Returned HTTP status code: $legacy_path_http_code."
+                    ;;
+                esac
                 rm -f "$zip_path" 2>&1 && say_verbose "Temporary zip file $zip_path was removed"
             fi
         fi
     fi
 
     if [ "$download_failed" = true ]; then
-        say_err "Could not find/download: \`$asset_name\` with version = $specific_version"
-        say_err "Refer to: https://aka.ms/dotnet-os-lifecycle for information on .NET Core support"
+        if [[ "$primary_path_http_code" = "404" && "$legacy_path_http_code" = "404" ]]; then
+            say_err "Could not find \`$asset_name\` with version = $specific_version"
+            say_err "Refer to: https://aka.ms/dotnet-os-lifecycle for information on .NET Core support"
+        else
+            say_err "Could not download: \`$asset_name\` with version = $specific_version:"
+            if [ "$primary_path_http_code" != "404" ]; then
+                say_err "$primary_path_download_error_msg"
+            fi
+            if [ "$legacy_path_http_code" != "404" ]; then
+                say_err "$legacy_path_download_error_msg"
+            fi
+        fi
         return 1
     fi
 
     say "Extracting zip from $download_link"
-    extract_dotnet_package "$zip_path" "$install_root"
```

Comment:
Is this change relevant to PR?

Reply:
Yes, if the script goes further it will end up with the "failed to install with an unknown error" message, which is confusing and wrong. The "extraction failed" message already explains the problem and we should finish on it. So, this change also concerns error messages.